sample/src/main/java/com/cloudinary/android/sample/app/ResourcesAdapter.java

```diff
@@ -261,7 +263,11 @@ private void setProgress(ResourceViewHolder holder, ResourceWithMeta resourceWit
 
         String text = holder.statusText.getContext().getString(R.string.uploading, progressStr);
         SpannableString spannableString = new SpannableString(text);
-        spannableString.setSpan(new ForegroundColorSpan(holder.statusText.getContext().getResources().getColor(R.color.buttonColor)), text.indexOf(progressStr), text.length(), 0);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            spannableString.setSpan(new ForegroundColorSpan(holder.statusText.getContext().getColor(R.color.buttonColor)), text.indexOf(progressStr), text.length(), 0);
```

Comment:
why don't you just extract `holder.statusText.getContext().getColor(R.color.buttonColor))` to a variable ? and reuse the rest of the code?

Reply:
Becase getColor from getResources is deprecated, and from getContext is supported in Android M+